### PR TITLE
docs: Update Readme

### PR DESCRIPTION
**Pull request by itleigns**
## Context
We can not find details about `sd-cmd` such as command options from the readme.

## Objective
We added explanations for `publish`, `validate`, `promote` and `removeTag` and added explanations of command options and examples for each feature.

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,12 +13,16 @@ $ go build -a -o sd-cmd
 ```
 
 ### Execute
+We execute a command. The command arguments can be specified by the following `arguments`.
```
Comment (yoshwata) [inline README.md:16]:
```suggestion
Executing a published command. The arguments for published command can be specified by the following `arguments`.
```

File: README.md [hunk 2]
```diff
@@ -27,6 +31,54 @@ It can be used in one of the following ways.
 - Use `-debug` of `--debug` option
 - Set `SD_CMD_DEBUG_LOG` environment variable to `true`
 
+### Publish
+We publish the command specified by a yaml.
```
Comment (yoshwata) [inline README.md:35]:
```suggestion
Publishing a command specified by a yaml.
```

File: README.md [hunk 3]
```diff
@@ -27,6 +31,54 @@ It can be used in one of the following ways.
...
+OPTIONS:
+   -f, --f    Specify the path of yaml to publish (default: sd-command.yaml)
+   -t, --t    Specify the tag given to the command (default: sd-command.yaml)
+
+EXAMPLE:
+   sd-cmd publish -f ./sd-command.yaml -t latest
+```
+
+### Validate
+We validate that a yaml is in the correct format for the command.
```
Comment (yoshwata) [inline README.md:49]:
```suggestion
Validating if a yaml is correct.
```

File: README.md [hunk 4]
```diff
@@ -27,6 +31,54 @@ It can be used in one of the following ways.
...
+
+OPTIONS:
+   -f, --f    Specify the path of yaml to validate (default: sd-command.yaml)
+
+EXAMPLE:
+   sd-cmd validate -f ./sd-command.yaml
+```
+
+### Promote
+We give the tag specified by `tag` to the version specified by `targetVersion`. The tag will be removed from the version which `tag` is originally assigned to. `targetVersion` can be set exact version or tag (e.g. 1.0.1, latest).
```
Comment (yoshwata) [inline README.md:62]:
```suggestion
We give the tag specified by `tag` to the version specified by `targetVersion`. The tag will be removed from the version which `tag` is originally assigned to. `targetVersion` can be set exact version or tag (e.g. 1.0.1, latest).

Giving a `tag` for a `targetVersion` of command. If a `tag` is already set to another version, that tag will be moved to `targetVersion`. `targetVersion` can be set exact version or tag (e.g. 1.0.1, latest).
```

File: README.md [hunk 5]
```diff
@@ -27,6 +31,54 @@ It can be used in one of the following ways.
...
+USAGE:
+   sd-cmd promote namespace/name targetVersion tag
+
+EXAMPLE:
+   sd-cmd promote foo/bar latest stable
+   sd-cmd promote foo/bar 1.0.1 stable
+```
+
+### Remove tag
+We remove the tag specified by `tag` from the version which `tag` is assigned to.
```
Comment (yoshwata) [inline README.md:73]:
```suggestion
Removing a `tag` from a version of published command.
```

File: README.md (hunk 3 above)
Comment (itleigns) [reply README.md:49]: [IMO] This seems the `validate` only checks the yaml is in correct format as yaml.

File: README.md (hunk 4 above)
Comment (itleigns) [reply README.md:62]: The following is right?
---
Giving a `tag` to a `targetVersion` of command.
---

File: README.md (hunk 3 above)
Comment (yoshwata) [reply README.md:49]: Is it better?
Validating if yaml is correct as sd-cmd format.

File: README.md (hunk 4 above)
Comment (yoshwata) [reply README.md:62]: It may be better.

File: README.md (hunk 3 above)
Comment (itleigns) [reply README.md:49]: It looks better for me.

**Comment by yoshwata** [comment]
Users use sd-cmd in the order of publish, execute, promote, removeTag, so I thought it would be better to follow the order of explanation.

**Comment by itleigns** [comment]
Most user only use `execute` so I think the order is better to be `execute`, `validate`, `publish`,  `promote`, `removeTag`.

**Review by yoshwata — APPROVED** [review]

**Review by ibu1224 — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -27,6 +31,54 @@ It can be used in one of the following ways.
...
+
+### Publish
+Publishing a command specified by a yaml.
+```bash
+USAGE:
+   sd-cmd publish [command options]
+
+OPTIONS:
+   -f, --f    Specify the path of yaml to publish (default: sd-command.yaml)
+   -t, --t    Specify the tag given to the command (default: sd-command.yaml)
```
Comment (tkyi) [inline README.md:55]:
```suggestion
   -t, --t    Specify the tag given to the command (default: latest)
```

Comment (itleigns) [reply README.md:55]: Thanks!

File: README.md [hunk 7]
```diff
@@ -13,12 +13,16 @@ $ go build -a -o sd-cmd
 ```
 
 ### Execute
+Executing a published command. The arguments for published command can be specified by the following `arguments`.
 ```bash
-$ sd-cmd exec [flags] namespace/name@version [arguments]
+USAGE:
+   sd-cmd exec [command options] namespace/name@version [arguments...]
```
Comment (wahapo) [inline README.md:19]:
```suggestion
   sd-cmd exec [options] namespace/name@version [arguments...]
```

File: README.md [hunk 8]
```diff
@@ -27,6 +31,54 @@ It can be used in one of the following ways.
...
 - Set `SD_CMD_DEBUG_LOG` environment variable to `true`
 
+### Validate
+Validating if yaml is correct as sd-cmd format.
+```bash
+USAGE:
+   sd-cmd validate [command options]
+
+OPTIONS:
+   -f, --f    Specify the path of yaml to validate (default: sd-command.yaml)
```
Comment (wahapo) [inline README.md:41]:
```suggestion
   -f, --f string    Specify the path of yaml to validate (default: sd-command.yaml)
```

File: README.md [hunk 9]
```diff
@@ -27,6 +31,54 @@ It can be used in one of the following ways.
...
+
+EXAMPLE:
+   sd-cmd publish -f ./sd-command.yaml -t latest
+```
+
+### Promote
+Giving a `tag` to a `targetVersion` of command. If a `tag` is already set to another version, that tag will be moved to `targetVersion`. `targetVersion` can be set exact version or tag (e.g. 1.0.1, latest).
+```bash
+USAGE:
+   sd-cmd promote namespace/name targetVersion tag
```
Comment (wahapo) [inline README.md:65]: Users will not be able to determine whether it is a fixed string or a variable string.
```suggestion
   sd-cmd promote namespace/name [targetVersion] [tag]
```

**Review by wahapo — APPROVED** [review]

**Comment by itleigns** [comment]
@tkyi
Could you please check this PR again?

**Review by tkyi — APPROVED** [review]

**Comment by itleigns** [comment]
Thank you!

**Pull request merged**